<launch>
    <arg name="dataset_dir" default="a2d2" />
    <arg name="split_dir" default="training" />
    <arg name="saving_path" default="a2d2_bev_images" />

    <arg name="camera_fov" default="110.0" />
    <arg name="planes" default="64" />
    <arg name="h_res" default="0.2" />
    <arg name="v_res" default="0.4" />
    <arg name="low_opening" default="-24.9" />
    <arg name="cell_size" default="0.05" />
    <arg name="grid_dim" default="140" />
    <arg name="grid_min_x" default="0" />
    <arg name="grid_max_x" default="35" />
    <arg name="grid_min_y" default="-35" />
    <arg name="grid_max_y" default="35" />
    <arg name="max_height" default="3.0" />
    <arg name="min_height" default="-9999.9" />
    <arg name="num_slices" default="1" />
    <arg name="height_threshold" default="0.10" />
    <arg name="cell_size_height_map" default="0.25" />
    <arg name="grid_dim_height_map" default="300" />
    <arg name="get_ground" default="false" />
    <arg name="remove_floor" default="false" />

    <node pkg="lidar_bev" type="offline_bev" name="offline_bev" output="screen">
        <param name="kitti_dir" value="$(arg dataset_dir)"/>
        <param name="split_dir" value="$(arg split_dir)"/>
        <param name="saving_path" value="$(arg saving_path)"/>
        <param name="camera_fov" value="$(arg camera_fov)"/>
        <param name="planes" value="$(arg planes)"/>
        <param name="h_res" value="$(arg h_res)"/>
        <param name="v_res" value="$(arg v_res)"/>
        <param name="low_opening" value="$(arg low_opening)"/>
        <param name="cell_size" value="$(arg cell_size)"/>
        <param name="grid_dim" value="$(arg grid_dim)"/>
        <param name="grid_min_x" value="$(arg grid_min_x)"/>
        <param name="grid_max_x" value="$(arg grid_max_x)"/>
        <param name="grid_min_y" value="$(arg grid_min_y)"/>
        <param name="grid_max_y" value="$(arg grid_max_y)"/>
        <param name="max_height" value="$(arg max_height)"/>
        <param name="min_height" value="$(arg min_height)"/>
        <param name="num_slices" value="$(arg num_slices)"/>
        <param name="height_threshold" value="$(arg height_threshold)"/>
        <param name="cell_size_height_map" value="$(arg cell_size_height_map)"/>
        <param name="grid_dim_height_map" value="$(arg grid_dim_height_map)"/>
        <param name="get_ground" value="$(arg get_ground)"/>
        <param name="remove_floor" value="$(arg remove_floor)"/>
    </node>

</launch>
